code/datums/move_intent/move_intent.dm

```diff
@@ -8,7 +8,20 @@
 	var/flags = 0
 	var/move_delay = 1
 	var/hud_icon_state
-	var/footstep_interval = 1
+
+/decl/move_intent/proc/can_be_used_by(var/mob/user)
```

Comment:
when is this check run?

We need to prevent people from sprinting when they're not standing up, this may be a good place for that deprending on when its called

If, as it seems, this is just used to prevent necromorphs from sprinting, it'd be better not to have this intent in their list at all

Reply:
> it'd be better not to have this intent in their list at all

This is how it currently works.
**can_be_used_by** was in that PR so I moved it without changes.